### PR TITLE
Rename master to main

### PR DESCRIPTION
**Pull request by youngnick**
Site has been tested with `make site-devel` seems, to be working.

For renames inside documentation, I've kept the rename to either:
- links to docs (ie `/docs/main/blah` instead of `/docs/master/blah`
- image tags etc.

Updates #2744.

## Important

We can't merge any PRs to `main` until this is merged.

Signed-off-by: Nick Young <ynick@vmware.com>

### CODE REVIEW

File: examples/prometheus/02-prometheus-configmap.yaml [hunk 1]
```diff
@@ -45,7 +45,7 @@ data:
       tls_config:
         ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
         # If your node certificates are self-signed or use a different CA to the
-        # master CA, then disable certificate verification below. Note that
+        # main CA, then disable certificate verification below. Note that
```
Comment (jpeach) [inline examples/prometheus/02-prometheus-configmap.yaml:48]: need to rephrase this

File: site/_resources/philosophy.md [hunk 2]
```diff
@@ -50,7 +50,7 @@ Contour rejects this position.
 
 In general, making something configurable implies taking no position on sensible values within its range.
 For example, many projects allow network buffer sizes to be configured arbitrarily, not just within the sensible range from a page to a few dozen pages, but often as low as 1 byte and as high as gigabytes.
-In providing the tools, but little or no guidance on how they should be used, infinitely tweakable knobs forces the burden of mastering the high dimensional configuration space onto the end users of the product.
+In providing the tools, but little or no guidance on how they should be used, infinitely tweakable knobs forces the burden of maining the high dimensional configuration space onto the end users of the product.
```
Comment (jpeach) [inline site/_resources/philosophy.md:53]: Undo this change.

File: site/docs/main/httpproxy.md [hunk 3]
```diff
@@ -1406,14 +1406,14 @@ Some examples of invalid configurations that Contour provides statuses for:
 - Contradictory header conditions on a route, e.g. a "contains" and "notcontains" condition for the same header and value.
 
  [1]: https://kubernetes.io/docs/concepts/services-networking/ingress/
- [2]: https://github.com/kubernetes/ingress-nginx/blob/master/docs/user-guide/nginx-configuration/annotations.md
+ [2]: https://github.com/kubernetes/ingress-nginx/blob/main/docs/user-guide/nginx-configuration/annotations.md
```
Comment (jpeach) [inline site/docs/main/httpproxy.md]: This is still master.

File: site/docs/v1.0.0/httpproxy.md [hunk 4]
```diff
@@ -1105,7 +1105,7 @@ Some examples of invalid configurations that Contour provides statuses for:
 
 
  [1]: https://kubernetes.io/docs/concepts/services-networking/ingress/
- [2]: https://github.com/kubernetes/ingress-nginx/blob/master/docs/user-guide/nginx-configuration/annotations.md
+ [2]: https://github.com/kubernetes/ingress-nginx/blob/main/docs/user-guide/nginx-configuration/annotations.md
```
Comment (jpeach) [inline site/docs/v1.0.0/httpproxy.md]: This is still master.

File: site/docs/v1.0.0/ingressroute.md [hunk 5]
```diff
@@ -1006,7 +1006,7 @@ Some examples of invalid configurations that Contour provides statuses for:
 - Root IngressRoute does not specify fqdn.
 
 [1]: https://kubernetes.io/docs/concepts/services-networking/ingress/
-[2]: https://github.com/kubernetes/ingress-nginx/blob/master/docs/user-guide/nginx-configuration/annotations.md
+[2]: https://github.com/kubernetes/ingress-nginx/blob/main/docs/user-guide/nginx-configuration/annotations.md
```
Comment (jpeach) [inline site/docs/v1.0.0/ingressroute.md]: This is still master.

File: site/docs/v1.0.1/httpproxy.md [hunk 6]
```diff
@@ -1105,7 +1105,7 @@ Some examples of invalid configurations that Contour provides statuses for:
 - Multiple header conditions of type "exact match" with the same header key.
 
  [1]: https://kubernetes.io/docs/concepts/services-networking/ingress/
- [2]: https://github.com/kubernetes/ingress-nginx/blob/master/docs/user-guide/nginx-configuration/annotations.md
+ [2]: https://github.com/kubernetes/ingress-nginx/blob/main/docs/user-guide/nginx-configuration/annotations.md
```
Comment (jpeach) [inline site/docs/v1.0.1/httpproxy.md]: This is still master.

File: site/docs/v1.0.1/ingressroute.md [hunk 7]
```diff
@@ -1006,7 +1006,7 @@ Some examples of invalid configurations that Contour provides statuses for:
 - Root IngressRoute does not specify fqdn.
 
 [1]: https://kubernetes.io/docs/concepts/services-networking/ingress/
-[2]: https://github.com/kubernetes/ingress-nginx/blob/master/docs/user-guide/nginx-configuration/annotations.md
+[2]: https://github.com/kubernetes/ingress-nginx/blob/main/docs/user-guide/nginx-configuration/annotations.md
```
Comment (jpeach) [inline site/docs/v1.0.1/ingressroute.md]: This is still master.

File: site/docs/v1.1.0/httpproxy.md [hunk 8]
```diff
@@ -1263,7 +1263,7 @@ Some examples of invalid configurations that Contour provides statuses for:
 - Multiple header conditions of type "exact match" with the same header key.
 
  [1]: https://kubernetes.io/docs/concepts/services-networking/ingress/
- [2]: https://github.com/kubernetes/ingress-nginx/blob/master/docs/user-guide/nginx-configuration/annotations.md
+ [2]: https://github.com/kubernetes/ingress-nginx/blob/main/docs/user-guide/nginx-configuration/annotations.md
```
Comment (jpeach) [inline site/docs/v1.1.0/httpproxy.md]: This is still master.

File: site/docs/v1.1.0/ingressroute.md [hunk 9]
```diff
@@ -1008,7 +1008,7 @@ Some examples of invalid configurations that Contour provides statuses for:
 - Root IngressRoute does not specify fqdn.
 
 [1]: https://kubernetes.io/docs/concepts/services-networking/ingress/
-[2]: https://github.com/kubernetes/ingress-nginx/blob/master/docs/user-guide/nginx-configuration/annotations.md
+[2]: https://github.com/kubernetes/ingress-nginx/blob/main/docs/user-guide/nginx-configuration/annotations.md
```
Comment (jpeach) [inline site/docs/v1.1.0/ingressroute.md]: This is still master.

File: site/docs/v1.2.0/httpproxy.md [hunk 10]
```diff
@@ -1301,7 +1301,7 @@ Some examples of invalid configurations that Contour provides statuses for:
 - Multiple header conditions of type "exact match" with the same header key.
 
  [1]: https://kubernetes.io/docs/concepts/services-networking/ingress/
- [2]: https://github.com/kubernetes/ingress-nginx/blob/master/docs/user-guide/nginx-configuration/annotations.md
+ [2]: https://github.com/kubernetes/ingress-nginx/blob/main/docs/user-guide/nginx-configuration/annotations.md
```
Comment (jpeach) [inline site/docs/v1.2.0/httpproxy.md]: This is still master.

File: site/js/jquery.matchHeight.js [hunk 11]
```diff
@@ -143,7 +143,7 @@
   *  plugin global options
   */
 
-  matchHeight.version = 'master';
+  matchHeight.version = 'main';
```
Comment (jpeach) [inline site/js/jquery.matchHeight.js]: no need to update this file.

File: examples/prometheus/02-prometheus-configmap.yaml (hunk 1 above)
Comment (youngnick) [reply examples/prometheus/02-prometheus-configmap.yaml:48]: This is a straight copy of an upstream thing, I think. I'll just revert.

**Comment by youngnick** [comment]
Okay, I think this should be good. @stevesloka, could you review for us and if you're okay to do the checks in #2744, merge? If you would prefer to wait until I'm online, no problems, I will check as soon as I'm online tomorrow.
